NWNXLib/Services/Commands/Commands.cpp

```diff
@@ -0,0 +1,128 @@
+#include "Services/Commands/Commands.hpp"
+#include <stdio.h>
+#include <dlfcn.h>
+
+namespace NWNXLib {
+namespace Services {
+
+static Commands *g_commands;
+
+Commands::Commands()
+{
+    g_commands = this;
+    RegisterCommand("nwnx_commands_test", [](std::string& s){ LOG_NOTICE("Selftest command. Args: '%s'", s.c_str()); });
+}
+
+Commands::~Commands()
+{
+    g_commands = nullptr;
+
+}
+
+bool Commands::RegisterCommand(const std::string& cmd, CommandFunc func)
+{
+    auto it = m_commandMap.find(cmd);
+    if (it == m_commandMap.end())
+    {
+        m_commandMap[cmd] = func;
+        LOG_INFO("Registering command '%s'", cmd.c_str());
+        return true;
+    }
+    else
+    {
+        LOG_WARNING("Command '%s' already registered", cmd.c_str());
```

Comment:
Perhaps this should be an exception, not a warning. That will stop NWNX from starting. This is how it works for the other services - so if there is a conflict between two plugins, the world stops until the users fixes their configuration. With that said, chat commands aren't exactly critical so a conflicting registration isn't the biggest deal. Hmm.

Reply:
Yeah, I didn't see it as end-of-the-world kind of thing. Maybe ERROR would be better, possibly an assert (breaks if you have a debugger), but most users won't be using these commands anyway..